src/net/sourceforge/kolmafia/request/FightRequest.java

```diff
@@ -7038,7 +7045,7 @@ private static boolean handleGrimstone(
   }
 
   private static final Pattern TRAINSET_MOVE =
-      Pattern.compile("^Your toy train moves ahead to the (.+?)\\.");
+      Pattern.compile("^Your toy train moves ahead to (?:(?:the)|(?:some)) (.+?)\\.");
```

Comment:
```suggestion
      Pattern.compile("^Your toy train moves ahead to (?:the|some) (.+?)\\.");
```
I don't think you need the inner capturing groups?